### DISCUSSION TITLE
String variable defined and used within a code block ?

### DISCUSSION THREAD

**Discussion by colindaven** [nextflow-io/nextflow#3764] [Q&A]
Hi,

**Edit - this method of saving to a file and reading from that file DOES actually work, I've tested it on different data now. It may not have worked on the first dataset as the algorithm is biased towards chromosome/seq ends, and the first data were only contigs. But - is there a better way to pass dynamic variables around a code block in nextflow. Perhaps environment variables ?**

Short version-  how should I define and use a string variable coming out of a bash command within a NF process code execution block ?

Long version -

I am trying to parse a string eg ATGAGAG from one output file (lines 1,2,3) and use it as input for another tool (command on the last line).
I have the following code, which works, but does not find the telomere (which should be present, as it was found in the same file) by the explore cmd on line 1.

Notes - the correct string is being parsed (line 2) and newline successfully removed on line 3.

Thanks.

```groovy
"""
    tidk_ubuntu_0.2.31 explore --length 7 --minimum 5 --maximum 12  $fasta > ${prefix}.telomeres.csv
    cut -f1  ${prefix}.telomeres.csv | head -n 2 | tail -n 1 > dynamic_telomere.txt
    cat dynamic_telomere.txt | tr -d '\n' > dynamic_telomere2.txt
    tidk_ubuntu_0.2.31 search --string <(dynamic_telomere2.txt) --output ${prefix}.dyn_telo --dir . --extension bedgraph $fasta
"""
```

**Comment by bentsherman** [comment]
I don't think Nextflow is involved here, since you are piping the output of one command into another. I would test this script by itself, without Nextflow, and make sure it works there first.
